### PR TITLE
138 add api endpoint for getting his own black card current one

### PR DESCRIPTION
**Pull request by daveking7**

### CODE REVIEW

File: src/main/java/ch/uzh/ifi/hase/soprafs22/controller/GameController.java [hunk 1]
```diff
@@ -252,4 +252,23 @@ public void submitWhiteCard(@RequestHeader(value = "authorization", required = f
...
       }
     }
+
+  /**
+   * Gets the current black card of a user or throws status exception
+   * @param token: token of the user
+   * @param userId: userId of the user
+   * @return: CardGetDTO
+   */
+    @GetMapping("/users/{userId}/blackCard")
```
Comment (TwoDigitsOneNumber) [inline src/main/java/ch/uzh/ifi/hase/soprafs22/controller/GameController.java:262]: i know in our application we only get at most 1 card, but i think we should still call it "blackCards" (as we did in the rest interface) to adhere to the standards mentioned in software engineering. What do you think? I'm fine to keep it in singular, but then we should adapt the rest interface, to reflect the design decision

File: src/test/java/ch/uzh/ifi/hase/soprafs22/controller/GameControllerTest.java [hunk 2]
```diff
@@ -1,6 +1,9 @@
 package ch.uzh.ifi.hase.soprafs22.controller;
 
-import ch.uzh.ifi.hase.soprafs22.entity.*;
+import ch.uzh.ifi.hase.soprafs22.entity.BlackCard;
+import ch.uzh.ifi.hase.soprafs22.entity.Game;
+import ch.uzh.ifi.hase.soprafs22.entity.User;
+import ch.uzh.ifi.hase.soprafs22.entity.WhiteCard;
```
Comment (TwoDigitsOneNumber) [inline src/test/java/ch/uzh/ifi/hase/soprafs22/controller/GameControllerTest.java:6]: good practice!

File: src/main/java/ch/uzh/ifi/hase/soprafs22/controller/GameController.java (hunk 1 above)
Comment (TwoDigitsOneNumber) [inline src/main/java/ch/uzh/ifi/hase/soprafs22/controller/GameController.java:262]: i know in our application we only get at most 1 card, but i think we should still call it "blackCards" (as we did in the rest interface) to adhere to the standards mentioned in software engineering. What do you think? I'm fine to keep it in singular, but then we should adapt the rest interface, to reflect the design decision.
Also: the api endpoint path is called users/{userId}/games/blackCards in the interface. I think we should keep the "/game" directory in there

File: src/test/java/ch/uzh/ifi/hase/soprafs22/controller/UserControllerTest.java [hunk 3]
```diff
@@ -518,7 +518,7 @@ void test_check_username_username_is_not_available() throws Exception {
    * can be processed
    * Input will look like this: {"name": "Test User", "username": "testUsername"}
    * 
-   * @param object
+   * @param wobject
```
Comment (TwoDigitsOneNumber) [inline src/test/java/ch/uzh/ifi/hase/soprafs22/controller/UserControllerTest.java:521]: is this a typo?

**Review by TwoDigitsOneNumber — CHANGES_REQUESTED** [review]
tests look good but api endpoint varies from rest interface. a design decision needs to be made

File: src/test/java/ch/uzh/ifi/hase/soprafs22/controller/GameControllerTest.java (hunk 2 above)
Comment (daveking7) [reply src/test/java/ch/uzh/ifi/hase/soprafs22/controller/GameControllerTest.java:6]: Props to my IDE

File: src/test/java/ch/uzh/ifi/hase/soprafs22/controller/UserControllerTest.java (hunk 3 above)
Comment (daveking7) [reply src/test/java/ch/uzh/ifi/hase/soprafs22/controller/UserControllerTest.java:521]: jup

**Review by JoMuelle — APPROVED** [review]
Looks good

**Pull request merged**